src/codeu/chat/util/Time.java

```diff
@@ -66,4 +66,10 @@ public String toString() {
 
   public static Time now() { return Time.fromMs(System.currentTimeMillis()); }
 
+  // Difference of end and start times.
+  // Ratio is the power of ten from seconds that the duration will be given in
+  // (e.i. ratio = -3 is milliseconds)
+  public static long duration(Time start, Time end, int ratio) {
+    return Math.round(Math.pow(10, -(3 + ratio)) * ((end.inMs() - start.inMs())));
```

Comment:
Please return duration in ms. It should be consistent with the rest of the Time class.

If you want to support more complicated use cases, then Duration should be its own class. But I don't think we'll need to re-implement all of [java.time.*](https://docs.oracle.com/javase/8/docs/api/java/time/package-summary.html) :smiley: